cogeo_mosaic/create.py

```diff
@@ -70,28 +65,81 @@ def create_mosaic(
         minzoom = max(minzoom)
 
     if maxzoom is None:
-        maxzoom = list(set([feat["properties"]["maxzoom"] for feat in results]))
+        maxzoom = {feat["properties"]["maxzoom"] for feat in features}
         if len(maxzoom) > 1:
             warnings.warn(
                 "Multiple MaxZoom, Assets have multiple resolution values", UserWarning
             )
 
         maxzoom = max(maxzoom)
 
-    quadkey_zoom = quadkey_zoom or minzoom
-
-    datatype = list(set([feat["properties"]["datatype"] for feat in results]))
+    datatype = {feat["properties"]["datatype"] for feat in features}
     if len(datatype) > 1:
         raise Exception("Dataset should have the same data type")
 
+    return create_mosaic_from_features(
+        features,
+        minzoom=minzoom,
+        maxzoom=maxzoom,
+        version=version,
+        quiet=quiet,
+        **kwargs,
+    )
+
+
+def create_mosaic_from_features(
```

Comment:
So any suggestion on what this should be called? I don't think it should be private. It's just that the above function is a higher-level overview for this.

Reply:
`from_features` sounds good to me (because it's in `mosaic.py`)

Reply:
Well currently this is `create.py`